stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -1007,8 +1007,7 @@ trait FlowOps[+Out, +Mat] {
    *
    * '''Cancels when''' downstream cancels
    */
-  @nowarn("msg=deprecated")
-  def mapConcat[T](f: Out => IterableOnce[T]): Repr[T] = statefulMapConcat(() => f)
+  def mapConcat[T](f: Out => IterableOnce[T]): Repr[T] = via(new MapConcat(f))
```

Comment:
Maybe not the correct opinion : 
(1) Does javadsl require similar modifications? maybe refer to `org.apache.pekko.stream.javadsl.Flow#statefulMapConcat`
(2) can we use object or case object for `MapConcat`, it is refered to `org.apache.pekko.stream.impl.fusing.MapAsync`.

Reply:
1. the javadsl's mapconcat is using scaladsl's mapconcat
2. case class will make the jar bigger and we have no pattern matching here.